### PR TITLE
feat(react): upgrade downshift dependency to v5

### LINKED ISSUE #2539: DropdownV2 downshift memory leak
## Detailed description

This applies to any component using downshift.

If the component (specifically DropdownV2 in my case) is focused, when the component is unmounted and dismissed after the submit, in a modal for example, react warns `Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application.`

## Steps to reproduce the issue

1. Display a modal with DropdownV2
2. Select an option in the dropdown
3. While the dropdown is still in focus, click the primary button submit in the modal
4. This will close the modal and display the memory error (only if the component is in focus when submitting)

## Additional information

This seems to have been a known error and solved in version 2.0.7 of downshift. Currently carbon-components-react seems to have ^1.31.14 as the downshift dependency version.
https://github.com/downshift-js/downshift/issues/476

### LINKED ISSUE #2628: Update downshift to the latest version ^2.0.0
Will need to refactor components that utilize it.

### LINKED ISSUE #3416: AVT 3 - React Combobox Component JAWS screenreader issues
## Environment
Windows 10 with JAWS 18
Firefox Quantum 68.0

## Description
1. On page load JAWS reads the title, optional text and listbox.
2. Tab to the Combobox and JAWS announces "Filter... Use arrow keys to navigate or type a value. Arrow keys work but typing stops after a couple of letters. 
**Expected Result:** User can type an option.
3. Press Tab focus skips the X icon.
**Expected result:** X icon receives focus.
3. Press the Space key the menu opens.
4. Press the Down arrow and focus moves out of the Combobox.
**Expected result:** the arrow keys traverse up and down the list.

### LINKED ISSUE #3419: AVT 3 - React Combobox macOS VoiceOver screenreader issues
## Environment
macOS 10.14.5 with VoiceOver
Chrome Version 75.0.3770.100 (Official Build) (64-bit)

## Description
1. Control-Option-Shift-Down Arrow to the main web content
2. Control-Option-Right Arrow. VO announces the Combox title and that it's a text element.
3. Control-Option-Right Arrow. VO announces the optional helper test and that it's a text element.
4. Control-Option-Right Arrow to the ComboBox. VO announces that it's a listbox with 0 items selected. To begin interacting press Control-Option-Shift-Down Arrow.
5. VO announces that you are in a listbox with 0 items selected. Open menu pop-up button 1 of 1, but the menu remains closed.
**Expected Result:** VO should also announce the "Filter..." text
6. Press Space and VO announces "5 results are available use the Up and Down Arrow keys to navigate. Arrow keys move the focus up and down through the list. 
7. Press Space on Option 2 to select it and VO announces, "You are currently on a menu inside a pop-up button. Type text. To display a list of choices press Control-Option-Space." Option 2 is selected and the menu list closes. 
8. Tab to the X icon and VO announces, "You are currently on a button inside of a pop-up button. To click the button press Control-Option-Space.
**Expected Result:** VO should also announce the purpose of the button.
9. Navigate and selected Option 2 again (it was cleared in the last step) and tab away from the Combobox.
10. Tab back to the Combobox and VO announces "Option 2 Option 2 selected."
Note: Did not expect to hear Option 2 announced twice.

### LINKED ISSUE #4373: AVT 3 Dropdown - VO does not announce the option after it is selected or the options position in the list..
## Environment
macOS Mojave version 10.14.6
Chrome Version Version 77.0.3865.90
Carbon v10 - React

## Detailed Description
Go to the React Component 
Start VO on macOS
1. Tab to the Dropdown, VO announces that it Open Menu pop-up 1 of 1.
**Expected Result** - VO announces the label "Option 1" and that it is a pop-up button.
2. Use the Enter key to open the dropdown and move to the next option. VO announces the option, but does not announce the position (i.e. 1 of 4, 2 of 4, etc).
**Expected Result**  - VO announces the Option # and position. 
3. Tab away from the dropdown and then tab back. VO announces, "Open Menu pop-up".
**Expected Result**- VO announces the that it is a menu pop-up and the current selection. 

See [ARIA Authoring Practices Dropdown Listbox example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html).

### LINKED ISSUE #5211: [Dropdown] Menu items are not read by JAWS 2019/NVDA
_this is a placeholder issue I'll flesh out more fully when I get the chance (definitely by tomorrow)_

Testing **React** Dropdown with NVDA and JAWS 2019 on Firefox latest for Windows 10 and after opening the Dropdown menu no information was given ("5 options" or whatever) and when focusing the various options they were not read and selections were not read. 

Suggest researching Downshift existing issues as this could be a known bug or problem with our implementation

### LINKED ISSUE #5335: [IE11] Visual bugs with SVG placement in combobox and text input validation
Two SVG visual bugs in IE11

Combobox
Screen Shot 2020-02-11 at 3 40 10 PM

TextInput
Screen Shot 2020-02-11 at 3 37 10 PM

### PR DESCRIPTION
**Pull request by dakahn**
Closes #2628
Closes #5335
Closes #2539
Closes #4373 
Closes #5211
Closes #3419
Closes #3416

Upgrade downshift dependency from v1 to v5 so we could take advantage of improved accessibility and a `useSelect` hook allowing Windows screen readers to enter forms mode appropriately.

### Changed:
-  packages/components/src/components/button/_mixins.scss
    - _added disabled styling_ 
- packages/components/src/components/dropdown/_dropdown.scss
    - _changed targeted triggering element to a `button`_
    - _added `button-reset`_
    - remove redundant/unneeded aria
- packages/components/src/components/list-box/_list-box.scss
    - changed targeted role from `combobox` to `type='text'`
- packages/react/src/components/Dropdown/Dropdown-test.js
    - changed label lookup from a `span` to a `label`
- packages/react/src/components/Dropdown/Dropdown.js
    - refactored for React Hooks
    - added Downshift's `useSelect` hook
    - removed redundant/unneeded aria
    - changed `ListBox.Field` to `button` element
- packages/react/src/components/ListBox/ListBox.js
    - added `React.ForwardRef`
    - remove `innerRef` hook
    - remove redundant/unneeded aria
- packages/react/src/components/ListBox/ListBoxMenu.js
    - added `React.ForwardRef`
- packages/react/src/components/ListBox/__tests__/ListBox-test.js
    - remove tests for `innerRef`
- packages/react/src/components/ListBox/test-helpers.js
    - add lookup for `role="combobox"`
    - add lookup for `aria-haspopup="listbox"`
    - remove helpers for `aria-expanded` and `aria-haspopup`
- packages/react/src/components/MultiSelect/FilterableMultiSelect.js
    - change deprecated `getButtonProps` to `getToggleButtonProps`
    - remove `innerRef`
- packages/react/src/components/MultiSelect/MultiSelect.js
    - refactored for React Hooks
    - added Downshift's `useSelect` hook
    - removed redundant/unneeded aria
    - changed `noop` value from `undefined` to `{}`
    - refactor deprecated `stateChangeTypes` to new change types for keyboard/mouse interactions
    - changed `ListBox.Field` to `button` element
- packages/react/src/internal/Selection.js
    - add `useCallback` `useState` hooks
    - add `useSelection` function
- packages/react/src/prop-types/childrenOfType.js
    - add early return if there is not child element
- packages/react/src/tools/createPropAdapter.js
    - add `createPropAdapter` for handling deprecated props

### Testing:
- Check for DAP errors on open and closed states for all dropdown components and their variants
    - MultiSelect
    - FilterableMultiselect
    - Dropdown
    - Comobox
_(if you find errors cross check for existing issues in our backlog)_
- Confirm issues are closed by this PR for:
   - VoiceOver for macOS on Safari latest
   - JAWS 2019^ on Windows 10 for Firefox and Chrome latest
   - NVDA on Windows 10 for Firefox and Chrome latest

### non-blocking known issues:
- "initial DOM focus should be on its textbox when a combobox receives focus" DAP error on Combobox is unrelated to these changes

### CODE REVIEW

File: packages/react/src/tools/createPropAdapter.js
Comment (joshblack) [inline packages/react/src/tools/createPropAdapter.js]: Maybe something like:

```js
/**
 * Create an adapter that converts an object of props with potentially deprecated
 * prop names to the replacement prop names in a newer version. Useful for guarding
 * against breaking changes when a prop has been renamed
 *
 * @param {Array} spec - an array of options which specify a text or regex
 * matcher alongside a replacement if there is a match
 * @returns {Function}
 */
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Left a quick nit for the function annotation, would we want to add a quick test for `mapDownshiftProps`?

**Review by tw15egan — DISMISSED** [review]
Noticing a couple of issues:

**Combobox**
- [x] Combobox `light` prop does not seem to work

~**Dropdown**~
~- [ ] [Inline variant] option bottom border is a bit short on the right side~
WIll be fixed via #5389

~**MultiSelect**~
~- [ ] [Inline variant] option bottom border is a bit short on the right side~
Will be fixed via #5389

**Comment by tw15egan** [comment]
Few comments but otherwise looks great, thanks for taking this on! Has been in the backlog for quite some time... https://github.com/carbon-design-system/carbon/issues/2628 😂

**Comment by dakahn** [comment]
@tw15egan great catch on the light prop not applying 👏 It looks like in our styling we were grabbing `role="combobox"` which isn't there anymore after the v5 update and should be updated to `type="text"`. I'll push up a fix.

Screen Shot 2020-02-18 at 11 52 39 AM

As for the other two bugs you mentioned -- it seems like those may have been introduced in an earlier commit? I'm seeing those on [the netlify preview](https://carbon-components-react.netlify.com/?path=/story/dropdown--default) :hurtrealbad:

**Comment by abbeyhrt** [comment]
I made an issue to look into the bottom borders being cut off! #5389 I'm gonna try and figure out where this got introduced since it's coming from what will be released in 10.10

**Comment by aledavila** [comment]
On default dropdown. when tabbing to the dropdown it announces "you are currently on a popup button to display a list of option press control-option-space" but that doesn't work you have to press space or enter to open the menu. When opening it and tabbing into the options this is the only thing that repeatedly announces in VO. It repeats it 5 times. 

Screen Shot 2020-02-18 at 12 44 34 PM

Once you choose an option it just repeats the same as above and never says the option you chose. 

Same things happens with the other dropdown options.

**Comment by dakahn** [comment]
Seeing some DAP errors being generated by the upgrade as well -- addressing those as we speak 👨‍🎤

**Comment by dakahn** [comment]
First thanks to Carolyn MacLeod for the assist today and yesterday 🙏 

So where we are now is -- we've got the menu items speaking again 🎉, but have two remaining DAP errors:

2020-02-21 19_15_07-Microsoft Edge


The first one has to do with Downshift _still setting a bunch of aria we don't need_ on elements that are explicitly labeled using `aria-label`. According to their docs this shouldn't happen, but I haven't been able to find out what it is about the structure of our dropdown that is causing their `...getMenuProps({ 'aria-label': ariaLabel)` trick they mention [here ](https://github.com/downshift-js/downshift#gettogglebuttonprops) to not work. 

Then there's that second DAP error -- might be one for another PR honestly since it'll take some heavy lifting to make this component not use a button/text input as it's dropdown container and fall more in line with WAI-Aria exapmles. 😞

**Comment by dakahn** [comment]
re-requesting some reviews. Not 100% but close enough for some feedback for sure 👍

File: packages/components/src/components/dropdown/_dropdown.scss
Comment (joshblack) [inline packages/components/src/components/dropdown/_dropdown.scss]: Style nit: make sure to un-nest selectors

```scss
.#{$prefix}--dropdown button {
  // ...
}
```

File: packages/react/src/components/ComboBox/ComboBox.js
Comment (joshblack) [inline packages/react/src/components/ComboBox/ComboBox.js]: If we pass in `aria-label` to `getMenuProps` do they return it so we don't have to set `aria-label` above?

File: packages/react/src/components/Dropdown/Dropdown.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown.js:176]: Style nit: should be after the Dropdown function declaration

File: packages/react/src/components/Dropdown/Dropdown.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown.js]: Style nit:

```suggestion
function Dropdown(props) {
```

`export default` should come after `propTypes` and `defaultProps`

File: packages/react/src/components/Dropdown/Dropdown.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown.js]: Style-nit: prefer function declarations to arrow functions for handlers

```suggestion
  function handleOnChange({ onChange }, selectedItem) {
```

File: packages/react/src/components/Dropdown/Dropdown.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown.js]:
```suggestion
    if (onChange) {
      onChange({ selectedItem });
    }
```

Would it be this? or `if (props.onChange)` ?

**Review by joshblack — COMMENTED** [review]
Super awesome to see the changes so far!

**Comment by abbeyhrt** [comment]
Looking at this now, I'm getting a couple of DAP violations on an open Combobox: 
Screen Shot 2020-03-19 at 2 47 31 PM
Just one issue on an open dropdown:

**Comment by abbeyhrt** [comment]
This is a doozy, I do not envy you @dakahn. Thanks fo much for doing it, hopefully we'll get a lot of a11y wins out of it! 
Looking at it now and I'm getting a couple of DAP violations that I'm just gonna dump here: 
For an open Combobox: 
Screen Shot 2020-03-19 at 2 47 31 PM
From an open Dropdown: 
Screen Shot 2020-03-19 at 2 49 50 PM
And then these two from an open or closed Multiselect: 
Screen Shot 2020-03-19 at 2 54 26 PM

I'm not sure if it needs to be but also noticed that the focus for the Dropdown list of options isn't styled: 
Screen Shot 2020-03-19 at 2 48 57 PM

File: packages/react/src/tools/createPropAdapter.js
Comment (joshblack) [inline packages/react/src/tools/createPropAdapter.js:1]: Make sure to add license banner to start of file 👍

File: packages/components/src/components/dropdown/_dropdown.scss
Comment (joshblack) [inline packages/components/src/components/dropdown/_dropdown.scss]: Should we use this selector to target it:

```scss
.#{$prefix}--dropdown .#{$prefix}--list-box__field {
  // ...
}
```

As an aside, `prefix` is missing the `$` in front 👍

File: packages/react/src/components/Dropdown/Dropdown.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown.js:46]: Are we forwarding `downshiftProps` anywhere?

**Comment by dakahn** [comment]
For the _"application-level control for selection"_ story for Combobox I'm also seeing a DAP error for **"initial DOM focus should be on its textbox when a combobox receives focus"** -- and yeah that story seems totally busted. I'm unable to type into the text box in Chrome latest on Windows 10. Seems to be a pre-existing problem. Checked both our react.carbon and netlify links and they were both bugged 🐛

**Comment by abbeyhrt** [comment]
@dakahn I only get the DAP violations when I test Multiselect and Combobox when their options menu is open and visible, but none when it's closed. It can take some fiddling to get DAP to run and not close the menu. 
Screen Shot 2020-03-24 at 12 40 36 PM

**Comment by dakahn** [comment]
@abbeyhrt cleared the mentioned DAP errors for combobox with the open menu save for one
**"Initial DOM focus should be on it textbox when a combobox receives focus"**
Which -- well focus _is_ on the textbox when the combobox receives focus, so let's ignore that one for now. 

Worth noting is that this error only pops up _when the menu dialog is open_. 😕

File: packages/components/src/components/list-box/_list-box.scss
Comment (emyarod) [inline packages/components/src/components/list-box/_list-box.scss:686]: this selector change is breaking light variant styles in the filterable multiselect, so we can either change the role on the input field in the filterable multiselect or change the style rule to apply to both selectors

**Comment by aledavila** [comment]
So far for macOS I can't get multiselect to read out the options or focus on them on VO 

Screen Shot 2020-03-25 at 4 26 09 PM

**Comment by dakahn** [comment]
MultiSelect is busted and is currently mid-overhaul -- refactoring to `useSelect`. Expect that code to land late tomorrow morning/lunch time yall 🙏 

(laughs at self from future)

**Comment by aledavila** [comment]
@dakahn thanks for the update ! 

Dropdown issues: 
closes #4373 

Windows on Firefox
closes #5211 

Windows on Edge
JAWS - cannot navigate with arrow keys down the list. If I hover over them with a mouse it announces the list item and position like its supposed to. 
NVDA - works fine

**Comment by aledavila** [comment]
Combobox on Mac
Closes #3419  

On Windows in Firefox 
Closes #3416 
with the exception of not being able to activate the close button. Navigates to it but it does not clear the selection

**Comment by silviuaavram** [comment]
Hey guys. Let me know if you need assistance in migrating. I hope you are considering to use our hooks for select, combobox and (soon) multiple selection. Feel free to create issues on the Downshift repo for any issues, improvements etc. I look forward to improve the hooks and your Downshift experience through the collaboration! :)

File: packages/components/src/components/dropdown/_dropdown.scss
Comment (tw15egan) [reply packages/components/src/components/dropdown/_dropdown.scss]:
```suggestion
  .#{$prefix}--dropdown .#{$prefix}--list-box__field {
```

File: packages/components/src/components/button/_mixins.scss
Comment (tw15egan) [inline packages/components/src/components/button/_mixins.scss:87]: Seems like
Screen Shot 2020-04-02 at 12 34 38 PM
 `Dropdown` is not getting the proper text/icon colors when disabled:

File: packages/components/src/components/list-box/_list-box.scss
Comment (tw15egan) [inline packages/components/src/components/list-box/_list-box.scss:688]: On L139 / L154, could we switch that to something like 

```css
border-bottom-color: transparent;
```

This will prevent the 1px jump when toggling disabled styles on/off

File: packages/components/src/components/list-box/_list-box.scss
Comment (tw15egan) [inline packages/components/src/components/list-box/_list-box.scss:689]: Focus styles on multi select need to be added
Screen Shot 2020-04-02 at 12 41 47 PM

File: packages/components/src/components/dropdown/_dropdown.scss
Comment (tw15egan) [inline packages/components/src/components/dropdown/_dropdown.scss:69]: Inline dropdown not rendering properly 

Screen Shot 2020-04-02 at 12 32 54 PM

**Review by tw15egan — COMMENTED** [review]
I left a few comments on some visual issues I'm seeing.

**Comment by dakahn** [comment]
You're all awesome. First round of fixes are in. HAVE AT EM! ⚔️

File: packages/components/src/components/dropdown/_dropdown.scss
Comment (joshblack) [inline packages/components/src/components/dropdown/_dropdown.scss:335]:
```suggestion
    // TODO: remove in v11
    .#{$prefix}--dropdown-text,
    .#{$prefix}--list-box__label {
```

Might help to keep the previous selector around until next release to protect against breaking changes

File: packages/components/src/components/dropdown/_dropdown.scss
Comment (joshblack) [inline packages/components/src/components/dropdown/_dropdown.scss:341]:
```suggestion
    // TODO: remove in v11
    .#{$prefx}--dropdown__arrow,
    .#{$prefix}--list-box__menu-icon svg {
```

File: packages/components/src/components/list-box/_list-box.scss
Comment (joshblack) [inline packages/components/src/components/list-box/_list-box.scss:686]:
```suggestion
  // TODO: remove [role='combobox'] in v11
  .#{$prefix}--list-box input[role='combobox'],
  .#{$prefix}--list-box input[type='text'] {
```

File: packages/react/src/components/Dropdown/Dropdown.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown.js:8]:
```suggestion
/**	
 * Copyright IBM Corp. 2016, 2018	
 *	
 * This source code is licensed under the Apache-2.0 license found in the	
 * LICENSE file in the root directory of this source tree.	
 */
 
import React from 'react';
```

**Review by joshblack — CHANGES_REQUESTED** [review]
- [x] When hovering over a disabled dropdown, the cursor shows up as if it is interactive instead of `cursor: not-allowed`
- What does design think of this tension between the red and blue focus in our validation state

Screen Shot 2020-04-06 at 3 59 38 PM

- [x] `MultiSelect.Filterable` it seems like pressing `ESC` will cause it to say an item is selected and then not selected even though nothing is selected

demo

- [x] Seems like we have an extra X in `MultiSelect.Filterable` in Edge and IE11 with anything that's a combobox

image

- [x] Seems like selecting an option in a dropdown causes the height to become insanely large (similarly with multi-select) in IE11

demo

- [x] Trying to clear a multi-select doesn't work in IE11

demo

**Comment by tw15egan** [comment]
Not seeing any visual / interaction issues with `ComboBox` 

Few other things I'm noticing:

## Dropdown
- [x] Small jitter with Dropdown menu when it is opened 

dropdown-jitter
___


- [x] The Dropdown story has an inline style of `width: 300px`, which is causing the `inline` variant to render weird. Wondering if we should add a storybook override to set a larger width on the `inline` variant

Screen Shot 2020-04-10 at 11 35 24 AM


## Multiselect

- [x] Noticing the same jitter as Dropdown when the menu is opened

multiselect-jitter
___


- [ ] Do we have a different focus style on items that are focused and selected? The focus gets lost when going through selected items.

filterable-focus
___


- [ ] It looks like we lost the ability to focus on the `X` to clear the selection. Here is the current functionality. 

multiselect-clear
___


- [x] Hover styles still being applied on the `X` icon when disabled

disabled-multiselect
___

- [x] Filterable Multiselect `light` variant is not being applied (The menu styles are correct, just not the field itself)

- [x] Getting a bunch of DAP errors on FIlterable MultiSelect




Really awesome seeing this come together! 🎉

**Comment by dakahn** [comment]
@tw15egan It seems to me like that jitter is caused by onmousedown on the dropdown field which moves focus to the field but leaves the menu open. Then menu seems to close onmouseup

I don't see the jitter when using a keyboard to open/close the dropdown, but just want to confirm that on your end.

It could be doable to have the menu close onmousedown, but then accidental onmousedown clicks couldn't be cancelled by dragging the mouse out of the element.

**Comment by tw15egan** [comment]
@dakahn the jitter seems to be caused by an additional 2px border when focused. To fix this, we usually set a transparent border when the focus is inactive so that elements do not shift when the border is added on focus.

**Comment by dakahn** [comment]
Ah! ok, I guess we were referring to two different jitters lol

**Comment by dakahn** [comment]
@tw15egan
3wsnn5
@joshblack pointed out that it looks like we can't focus the X to clear the selection in latest. 

It also look like the hover styles still being applied to the filter tag on hover when it's disabled was a pre-existing bug as well, but I threw up a fix for that one real quick 👍

**Comment by dakahn** [comment]
@joshblack To make the button in a button work in IE11 it seems like if we change the descendant button to a `<button>` element instead of a div button we get the behavior expected. That said having an interactive descendant inside of a button goes against HTML5 spec. [I've opened up a ticket to get some feedback/discuss solutions](https://github.com/carbon-design-system/carbon/issues/5880)

I can go ahead an make the aforementioned changes though to at least get working functionality for mouse users on IE11 in the meantime

**Comment by joshblack** [comment]
@dakahn definitely your call. If it's something that's busted in latest feel free to just skip it and we can address it after this gets merged in with whatever the ideal is. If it's an easy fix in the interim then definitely makes sense to include it here 👀

**Comment by joshblack** [comment]
@dakahn seems like there may be some multi-select tests failing now 🤔 Just pushed up an update for snapshots btw

**Comment by asahlin** [comment]
I was informed that this issue covers the condition with the multiselect component and no visual indictor on focus / highlighted menu item when using the keyboard, details below.   Is this correct and is there an ETA for this pr to be available.

> Pre-condition: All items are checked.
> 
> When you expand the multi-select field and use the arrow keys on the keyboard to select, the keyboard focus indicator is not visible. We can not know which element has the keyboard focus.  All focus indicators should be highly visible and distinguishable.   In the screen shot below, focus is on the 2nd item, but there is no indication.

image

**Comment by dakahn** [comment]
@asahlin This PR does not cover that issue. Can you link me the ticket you're referencing and I can comment.

**Comment by dakahn** [comment]
@tw15egan @joshblack tests are corrected and we seem g2g if yall wanna give it another perusal 😸

**Comment by tw15egan** [comment]
Few things I'm noticing: 

### Dropdown
-  [x] `Top` menu orientation doesn't seem to be doing anything

### MultiSelect
- [x] `Top` menu orientation doesn't seem to be doing anything
- [x] The inline version seems to be off by 1px somewhere

Screen Shot 2020-05-05 at 9 00 22 AM

(Dropdown seems to be look correct)
Screen Shot 2020-05-05 at 9 00 29 AM

File: packages/react/src/components/Dropdown/Dropdown-story.js
Comment (joshblack) [inline packages/react/src/components/Dropdown/Dropdown-story.js]:
```suggestion
```

File: packages/react/src/components/Checkbox/Checkbox.js
Comment (joshblack) [inline packages/react/src/components/Checkbox/Checkbox.js]: Any reason for this change? Would have thought that the label would still refer to this node even if the DOM order is flipped

**Review by joshblack — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Checkbox/Checkbox.js
Comment (dakahn) [reply packages/react/src/components/Checkbox/Checkbox.js]: it was something to do with how it read. honestly, I can't recall and I can just change it back 😆 . There has been some good updates to checkbox while this PR was in flight anyway and I wouldnt want this change to make some newer change null or worse

**Comment by dakahn** [comment]
@tw15egan is it one pixel too much padding on the left side?

**Comment by tw15egan** [comment]
@dakahn it looks like it's coming from [this line](https://github.com/carbon-design-system/carbon/blob/6be11ccc12b4c6a44313c41e13cb010b920ae0cd/packages/components/src/components/list-box/_list-box.scss#L258).

I think we can fix this by keeping that line (needed to fix focus issue on normal Dropdowns) but adding in an inline-specific override like

```scss
  .#{$prefix}--list-box--inline .#{$prefix}--list-box__field {
    height: 100%;
  }
```

Last style issue I'm seeing, everything else is looking great!

**Review by tw15egan — APPROVED** [review]
Awesome work 👍 ✅ 🏄‍♂️ 🎉

**Review by joshblack — APPROVED** [review]

**Pull request merged**
